### ISSUE TITLE
implementation for pytorch gemma InifiniTransformer is copied without attribution

### ISSUE THREAD

**Issue by wompwompsquared** [mustafaaljadery/gemma-2B-10M#11]
The code for the model provided in this repository seems to be a copy of the repository linked below:

https://github.com/Beomi/InfiniTransformer/blob/main/infini_gemma/modeling_infini_gemma.py

Specifically, the GemmaInifniAttention and GemmaModel seem to be a direct copy, with comments removed, gradient checkpointing removed, and specific sections of the code slightly altered (especially in the aforementioned classes). 

The only actual difference other than slight code re-writings (e.g., flipping conditional statements, shuffling variable definition positions, turning multi-line statements into a single long line) seems to be that you forgot to add the rotary embeddings in the GemmaInfiniAttention class, replaced instances of segment variables with hidden_states. All of the variable names are identical.

What do the authors of this repository have to say in response? There doesn't seem to be anything new, and there is no mention of the original authors of the unofficial implementation. Not a very good look considering the recent llama3v incident...

**Comment by JamesClarke7283** [comment]
> The code for the model provided in this repository seems to be a copy of the repository linked below:
> 
> https://github.com/Beomi/InfiniTransformer/blob/main/infini_gemma/modeling_infini_gemma.py
> 
> Specifically, the GemmaInifniAttention and GemmaModel seem to be a direct copy, with comments removed, gradient checkpointing removed, and specific sections of the code slightly altered (especially in the aforementioned classes).
> 
> The only actual difference other than slight code re-writings (e.g., flipping conditional statements, shuffling variable definition positions, turning multi-line statements into a single long line) seems to be that you forgot to add the rotary embeddings in the GemmaInfiniAttention class, replaced instances of segment variables with hidden_states. All of the variable names are identical.
> 
> What do the authors of this repository have to say in response? There doesn't seem to be anything new, and there is no mention of the original authors of the unofficial implementation. Not a very good look considering the recent llama3v incident...

Indeed...
This repository does not even have a license attached in any form, as the berne convention dictates, Copyright is granted automatically(in almost every country) and defaults to 'All rights Reserved'. this raises many many issues, but i will highlight 2 , `#1`: anyone who uses this will have to assume its all rights reserved, `#2`: the original source, uses Apache2.0 which is very permissive, and only requires the notice be preserved and license attached.

the copyright owners(as the copytight notice states) are `Google Inc. HuggingFace Inc. team` , if they wanted to pursue this through legal action i would argue they would have a strong case.

@mustafaaljadery @akshgarg7  please resolve this promptly, i am surprised this hasn't been resolved already, at time of writing its been 266 days since this was reported, and commits have been made on this repo since.

I say this out of concern, i realise this is likely done in error and should be fairly easy to resolve.
